### ISSUE TITLE
MCHF calculations in Psi4?

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#1778]
I'm hoping to do atomic MCHF calculations with Psi4, which amount to small CASSCF calculations. However, I can't get DETCI to run the calculations for me.

Take the oxygen atom for example. The 3P state has 4 electrons on the P shell. DETCI correctly determines that there is 1 alpha string (since the shell is fully occupied) and 3 beta strings, but then it says that the CI space only has a single determinant and crashes.

Example inputs

```
molecule {
0 3
O
}

set basis cc-pVDZ
set reference rohf
set restricted_docc [2, 0, 0, 0, 0, 0, 0, 0]
set active [0, 0, 0, 0, 0, 1, 1, 1]
energy('casscf')
```
or a lower symmetry
```
molecule {
symmetry c2h
0 3
O
}

set basis cc-pVDZ
set reference rohf
set restricted_docc [2, 0, 0, 0]
set active [0, 0, 1, 2]
energy('casscf')
```

I guess the problem is that DETCI can't find a proper symmetry? Is there a way to circumvent this issue? @CDSherrill ?

**Comment by susilehtola** [comment]
I think the calculation would run, if there was a way to force DETCI to include **all** strings in the CI space. @dgasmith might also know?

**Comment by dgasmith** [comment]
Sorry, I haven't worked through the string/det part of the code much. This is indeed a @CDSherrill  question.

**Comment by CDSherrill** [comment]
This is one of those rare cases where the HF reference actually has some
influence.  (Usually, we say that the HF is "irrelevant" in a CASSCF
computation).  The issue is that the code can't determine what irrep
your target wavefunction is supposed to be, because you haven't given it
information about the orbital occupancy of the reference HF wavefunction.
I think if you specified docc [] and socc [] arrays for the reference
high-spin triplet, the code would have a much better chance to figure out
what you want.

**Comment by CDSherrill** [comment]
Hm, nope, that doesn't get around the problem.  I'll see if I can take a
look if I get a minute free later.

**Comment by CDSherrill** [comment]
I've looked at this a little more.

The issue is that DETCI was written assuming that one wants to find a
wavefunction of a definite irreducible representation.  So, within D2H, or
C2H, your 3 determinants have different symmetries.  I think the only
subgroup in which they will have the same irrep is if it is run in C1.

So, I tried setting it up in C1. It runs, but it just gives the HF energy
back.  Why?  Well, even if we tell the program "C1", the system still has
the underlying atomic symmetry (D2H being the highest Abelian subgroup).
Under D2H symmetry, those 3 determinants won't mix, and they don't.

Of course, the *true* symmetry of the molecule is atomic symmetry, higher
than D2H.  But unfortunately we only have Abelian subgroups in Psi.

I am not sure offhand if one could force a mixing by deliberately breaking
the symmetry of the orbitals, or by using a different guess CI vector.  I'd
have to think about that a little more.  It might be that to do what you
are trying to do and get a fully-spin-adapted multi-configurational
wavefunction, one would need to use actual spherical harmonic atomic
orbitals, not the Cartesian representations p_x, p_y, p_z.
